# UMP12: Hersteller von Bestückungsmaschinen

## Lösung ILP in algebraischer Notation
Nachfolgendes Skript zeigt die Lösung des Modells als ILP, d.h. als ganzzahlig lineares Programm, ausgehend von einer Formulierung in **algebraischer Notation**. 


In [2]:
from UMP12_ILP_ALGEBRAIC_NOTATION import *

main()

Solution:
Objective value = 70000.0
X_0_0 = 40.0
X_0_1 = 0.0
X_0_2 = 0.0
X_0_3 = 40.0
X_0_4 = 30.0
X_1_0 = 0.0
X_1_1 = 0.0
X_1_2 = 110.0
X_1_3 = 0.0
X_1_4 = 0.0
X_2_0 = 0.0
X_2_1 = 130.0
X_2_2 = 0.0
X_2_3 = 10.0
X_2_4 = 0.0


Nun müssen die Ausgaben der Entscheidungsvariablen noch korrekt interpretiert werden. Die optimalen Transportkosten belaufen sich auf 70'000 CHF, die Lösung gestaltet sich wie folgt:

- Produktionsstandort 1 liefert
    - 40 Maschinen an Kunde 1
    - 30 Maschinen an Kunde 4
    - 30 Maschinen an Kunde 5
- Produktionsstandort 2 liefert
    - 110 Maschinen an Kunde 3
- Produktionsstandort 3 liefert
    - 130 Maschinen an Kunde 2
    - 20 Maschinen an Kunde 4


## Lösung ILP in algebraischer Notation mit Excel-Input
Nachfolgendes Skript löst nochmals dasselbe Modell als ILP in **algebraischer Notation**. Dieses Mal werden die Parameter jedoch aus einer Excel-Datei eingelesen. So soll demonstriert werden, wie grössere Problemstellungen über Dateien konfiguriert und eingelesen werden können.


In [ ]:
from UMP12_ILP_EXCEL_BASED import *

main()